File: VAL.py
import numpy as np
import matplotlib.pyplot as plt
from psychoacoustic_metrics import Tonality_Aures1985

# Define parameters
fs = 44100              # Sampling rate in Hz
duration = 100          # Duration in seconds
f0 = 1000               # Frequency of pure tone (Hz)
Lp = 60                 # Desired sound pressure level (dB SPL)
pref = 20e-6            # Reference pressure in Pa

# Create time vector
t = np.arange(0, duration, 1/fs)

# Generate sine wave with RMS level corresponding to 60 dB SPL
rms_target = pref * 10**(Lp / 20)
amp = rms_target * np.sqrt(2)
signal = amp * np.sin(2 * np.pi * f0 * t)

# Compute tonality using Aures 1985 model
result = Tonality_Aures1985(signal, fs=fs, LoudnessField=0, time_skip=0.5, show=True)

print("Mean Tonal Weighting (w_tonal):", np.mean(result['TonalWeighting']))
print("Mean Loudness Weighting (w_gr):", np.mean(result['LoudnessWeighting']))
print("Mean Tonality:", np.mean(result['InstantaneousTonality']))

# Print statistics
print("Tonality Statistics:")
for k, v in result.items():
    if isinstance(v, float) or isinstance(v, np.ndarray) and np.ndim(v) == 0:
        print(f"{k}: {v:.4f}")


# # Define parameters
# wav_file_path = "test_signal.wav"  # Path to a test WAV file
# fs_target = 44100                 # Target sampling frequency
# LoudnessField = 0                 # Free field
# time_skip = 0.5                   # Skip first 0.5 seconds
# show = True                       # Enable plots
# dBFS = 94                         # Reference full-scale SPL in dB

# # Generate a test WAV file (1 kHz sine wave at 60 dB SPL)
# fs_original = 48000               # Original sampling frequency
# duration = 5                      # Duration in seconds
# f0 = 1000                         # Frequency of pure tone (Hz)
# Lp = 60                           # Desired sound pressure level (dB SPL)
# pref = 20e-6                      # Reference pressure in Pa

# t = np.arange(0, duration, 1/fs_original)
# rms_target = pref * 10**(Lp / 20)
# amp = rms_target * np.sqrt(2)
# signal = amp * np.sin(2 * np.pi * f0 * t)

# # Save the signal as a WAV file
# from scipy.io.wavfile import write
# write(wav_file_path, fs_original, (signal * 32767).astype(np.int16))  # Save as 16-bit PCM

# # Run Tonality_Aures1985 with the WAV file input
# result = Tonality_Aures1985(wav_file_path, fs=fs_target, LoudnessField=LoudnessField, time_skip=time_skip, show=show, dBFS=dBFS)

# # Print results
# print("Mean Tonal Weighting (w_tonal):", np.mean(result['TonalWeighting']))
# print("Mean Loudness Weighting (w_gr):", np.mean(result['LoudnessWeighting']))
# print("Mean Tonality:", np.mean(result['InstantaneousTonality']))

# # Print statistics
# print("Tonality Statistics:")
# for k, v in result.items():
#     if isinstance(v, float) or isinstance(v, np.ndarray) and np.ndim(v) == 0:
#         print(f"{k}: {v:.4f}")

File: Original Matlab/psychoacoustic_metrics/Tonality_Aures1985/VAL.m
%% 1. PARAMETERS
fs       = 44100;             % Sampling rate in Hz
duration = 1.0;               % Signal duration in seconds
f0       = 1000;              % Tone frequency in Hz
Lp       = 60;                % Desired SPL in dB
pref     = 20e-6;             % Reference pressure in Pa

%% 2. TIME VECTOR
t = 0 : 1/fs : duration - 1/fs;  % Time vector

%% 3. SIGNAL GENERATION
rms_target = pref * 10^(Lp / 20);     % RMS amplitude for desired SPL
amp        = rms_target * sqrt(2);    % Peak amplitude for sine wave
signal     = amp * sin(2 * pi * f0 * t);  % Generate pure tone

%% 4. TONALITY CALCULATION
LoudnessField = 0;       % 0 = free field
time_skip     = 0.5;     % Skip first 0.5 seconds
show_plot     = true;    % Plot results

OUT = Tonality_Aures1985(signal, fs, LoudnessField, time_skip, show_plot);

%% 5. PRINT RESULTS
fprintf('\nTonality Statistics:\n');
fprintf('  InstantaneousTonality : %.3f\n', mean(OUT.InstantaneousTonality));
fprintf('  LoudnessWeighting     : %.3f\n', mean(OUT.LoudnessWeighting));
fprintf('  TonalWeighting        : %.3f\n', mean(OUT.TonalWeighting));